core/src/test/java/org/infinispan/distribution/groups/TransactionalGetGroupKeysTest.java

```diff
@@ -140,8 +140,7 @@ public void testGetGroupsWithConcurrentReplace() throws Exception {
 
       testCache.primaryOwner.put(key(1), value(-1));
 
-      if ((factory == TestCacheFactory.PRIMARY_OWNER || factory == TestCacheFactory.BACKUP_OWNER) &&
-            isolationLevel == IsolationLevel.READ_COMMITTED) {
+      if (isolationLevel == IsolationLevel.READ_COMMITTED) {
```

Comment:
Ok, now I think I really got it :)

I got confused by the `GetKeysInGroupCommand` name, I thought it returns just the keys and the originator then has to invoke a `GetAllCommand` to get all the values (that are not already in the context). But since we're including those values in the response anyway, it makes sense to update the context entries with them.